index.js

```diff
@@ -130,6 +130,15 @@ bot.on('message', function (message) {
         return;
       }
       switch (lCommand) {
+        case 'random':
+          let results = scraps.find({ 'botMessageId': { '$nin': [null, 'IN_PROGRESS'] } });
```

Comment:
What does this bit do? `null, 'IN_PROGRESS'`

Reply:
I see. It's some state that's there already. /shrug

Reply:
From what I can see, it filters out snaps that are in the progress of being saved to the db and so don't have an id yet. See line 496-506 ish